homeassistant/components/upnp/sensor.py

```diff
@@ -54,10 +60,30 @@
         "name": f"{DATA_PACKETS} sent",
         "unit": DATA_PACKETS,
         "unique_id": PACKETS_SENT,
+        "format": "d",
         "derived_name": f"{DATA_RATE_PACKETS_PER_SECOND} sent",
         "derived_unit": DATA_RATE_PACKETS_PER_SECOND,
         "derived_unique_id": "packets/sec_sent",
     },
+    ROUTER_IP: {
+        "device_value_key": ROUTER_IP,
+        "name": "IP",
+        "unit": None,
+        "unique_id": ROUTER_IP,
+    },
+    ROUTER_UPTIME: {
```

Comment:
This should be disabled by default
https://developers.home-assistant.io/docs/entity_registry_disabled_by/#integrations-offering-options-to-control-disabled_by

Reply:
Thx!!! Updated accordingly, please check if I did it right. :-)

Reply:
Not really, everything will be enabled by default now (which is the default as well)

Reply:
Not following, for the UpTime sensor I provide a boolean to determine if it should be enabled or not. All others I do not provide anything and then yes, it is enabled by default but right now it really is just uptime that is not always supported.

Reply:
<https://github.com/home-assistant/core/pull/54780/files#diff-1f0fc2dfe3626e3495b8aee8d906009a32c1df3dc753b46b7fda9ed753a1113aR121> Is this what you mean?
if so, use `enabled=False`  or `False` instead.

Reply:
So disabling entities now if they ended up return None which would mean not available/supported by the upnp device.

Reply:
Those should not be added in the first place.

Reply:
I'm not following. What do you mean they should not be added in the 1st place? At first I was not adding Uptime if not available/support by the upnp device but instead should add it but disabled.
If the device does not support something, should we add disabled or not add at all?

Thx.

Reply:
Disabled is a good default for an uptime sensor.

But if we know an entity will never work, there is no need to even set that sensor up, here we have already done a data fetch before setting up the sensor, so we should add a if clause to the for look that add entities to check if this should be set up

Reply:
So the uptime sensor returns back how long the UPnP device has been up. However, not every UPnP device supports this (although per standard they should).
For example, my router supports it but my cable modem does not. So before it would create the entity for my router but not for my cable modem.
Right now it would create them for both, but the entity for the cable modem is unavailable.

So I will add a check in the for loop then to confirm that particular sensor is available from the UPnP device and if it is not then not create the entity.